src/orange/universes/detail/SurfaceFunctors.hh

```diff
@@ -71,6 +70,13 @@ struct CalcSafetyDistance
     template<class S>
     real_type operator()(S&& surf)
     {
+        if (!S::simple_safety())
+        {
+            // Not a surface that satisfies our simplifying constraints: return
+            // a conservative answer.
+            return 0;
+        }
```

Comment:
I didn't try to understand the whole procedure yet, but a zero returned certainly looks odd. Any quick tips for my understanding of the logic of returning zero here?

Reply:
Since we can't calculate the actual safety distance number, any other number may result in an incorrect (dangerously large, for instance) safety distance. A return value of zero limits the MSC step to the shortest possible one. The only other safe approach would be to abort the simulation if the geometry has unsupported shapes.